spark/src/main/scala/geotrellis/spark/io/AttributeStore.scala

```diff
@@ -95,21 +115,40 @@ trait BlobLayerAttributeStore extends AttributeStore {
     cacheRead[JsValue](id, Fields.metadataBlob).asJsObject.fields(Fields.header).convertTo[H]
 
   def readMetadata[M: JsonFormat](id: LayerId): M =
-    cacheRead[JsValue](id, Fields.metadataBlob).asJsObject.fields(Fields.metadata).convertTo[M]
+      cacheRead[JsValue](id, Fields.metadata).asJsObject.fields(Fields.metadata).convertTo[M]
+
+  def readKeyIndex[K: ClassTag](id: LayerId): KeyIndex[K] = {
+    layerType(id) match {
+      // TODO: Find a way to read a single KeyIndex from the COGMetadata
+      case COGLayerType => throw new COGLayerAttributeError("schema", id)
```

Comment:
Can we make this function work? Should be not complicated: https://github.com/locationtech/geotrellis/blob/master/spark/src/main/scala/geotrellis/spark/io/cog/COGLayerMetadata.scala#L50

Reply:
The issue with implementing this method for COG layers is that all its saved attributes are associated with zoom level `0`, so we'd have to do some additional work before the `KeyIndex` can be read. If that's okay then should be able to make this method work.

Reply:
@pomadchin I don't see an alternative alternative as of right now. However, I think that once we start refactoring `AttributeStore` it'll be some something we can do.

I'm not sure how much this method would be needed for a COG layer. Though, I don't think it's that much. That's why I opted to just throw an exception for right now.